packages/collector/src/agentConnection.js

```diff
@@ -486,3 +490,42 @@ function logLargeSpans(spans) {
     )}`
   );
 }
+
+/**
+ * @typedef {InstanaBaseSpan} Span Span
+ * @property {number} [k] kind
+ */
+
+/**
+ * Returning type for the function getSpanLengthInfo
+ * @typedef {Object.<string, number>} CountBySpanType
+ */
+
+/**
+ * @param {Span[]} spans
+ * @returns {CountBySpanType}
+ */
+function getSpanLengthInfo(spans) {
```

Comment:
S:

As we affect the customers execution time with our tracer, we could add:

```
if (logger.level !== bunyan.DEBUG) return {}
```

Syntax needs to be double checked.

Reply:
Good suggestion. That would help improve the performance of our tracer for customer